### ISSUE TITLE
Suggestions with wrong distance

### ISSUE THREAD

**Issue by alsora** [Lundez/JavaSymSpell#3]
Hi!

I am trying your java version of SymSpell.
It works really well, however I have noticed a very strange behaviour.

I am using the provided demo, with frequency_dictionary_en_500_000.txt as dictionary ( find it [here](https://github.com/wolfgarbe/SymSpell/blob/master/SymSpell.Benchmark/test_data/frequency_dictionary_en_500_000.txt)  on SymSpell page )

If I type as input "the", only this word, without spaces, the suggestions are the following:
```
Enter input:   
the   
Lookup suggestion: ﻿the 1 6801236995 
Lookup suggestion: he 1 588605295
Lookup suggestion: they 1 326802098
Lookup suggestion: them 1 178377546
Lookup suggestion: she 1 140368041
Lookup suggestion: then 1 108733059
Lookup suggestion: thy 1 11355619
Lookup suggestion: thee 1 7761536
Lookup suggestion: tho 1 4200344
Lookup suggestion: tie 1 1819888
LookupCompound: ﻿the
```

The first is correctly the word itself, by the way it has a distance of 1 from the input word... It should be 0!

This is an isolated case, as with other correct words, I'm correctly getting a 0 distance.
Do you have any idea why this happens?

**Comment by Lundez** [comment]
Hello. Thanks for finding this! I'll check it out later this week. :)

**Comment by Lundez** [comment]
Hey alsora!

Sorry for the late answer.
This was a rather weird but fun bug. It all came down to Windows and their formatting on UTF-8 files. They add a "BOM" which is a char in the beginning of the file. This char wasn't removed by .trim() in my first try. I ended up resaving the dictionary using Notepad++ and removing the BOM.

I was helped by this issue to pin it down when I found out that the 4th (first) character was empty: https://stackoverflow.com/a/6543595/8329323

I'll close this issue now :)
